quickwit-search/src/service.rs

```diff
@@ -150,6 +150,10 @@ impl SearchService for SearchServiceImpl {
         &self,
         fetch_docs_request: FetchDocsRequest,
     ) -> crate::Result<FetchDocsResponse> {
+        let search_request = fetch_docs_request
+            .search_request
+            .ok_or_else(|| SearchError::InternalError("No search request.".to_string()))?;
```

Comment:
Would be nice to require the search request and the doc mapper only when snippeting is on.

Reply:
I found it makes the logic convoluted, but it has the advantage on send less data over the network when not needed.